ee/agent/storage/sqlite/keyvalue_store_sqlite.go

```diff
@@ -324,7 +327,20 @@ ON CONFLICT (name) DO UPDATE SET value=excluded.value;`
 		}
 		return nil, fmt.Errorf("deleting keys: %w", err)
 	}
-	defer rows.Close()
+	defer func() {
+		if err := rows.Close(); err != nil {
+			s.slogger.Log(context.TODO(), slog.LevelWarn,
+				"closing rows after scanning results",
+				"err", err,
+			)
+		}
+		if rows.Err() != nil {
+			s.slogger.Log(context.TODO(), slog.LevelWarn,
+				"encountered iteration error",
+				"err", err,
```

Comment:
should this capture the `err` from `rows.Err()`?

Reply:
Oops yes thank you!